vscode-antimony/src/server/stibium/stibium/ant_types.py

```diff
@@ -350,7 +395,210 @@ def get_comp(self):
+    
+    def to_string(self):
+        ret = ''
+        for paren in self.get_all_parens():
+            ret += paren.get_paren()
+        return ret
+
+    def check_rep(self):
+        for child in self.children:
+            assert isinstance(child, Parenthesis)
+            assert child.text == '(' or child.text == ')'
+            
+@dataclass
+class EventDelay(TrunkNode):
+    children: Tuple[Union[Sum, VarName], Keyword] = field(repr=False)
+    
+    def get_sum(self):
+        return self.children[0]
+    
+    def get_sum_text(self):
+        ret = ''
+        delay = self.get_sum()
+        if isinstance(delay, LeafNode):
+            ret += delay.text
+        else:
+            ret += delay.to_string()
+        ret += ' after'
+        return ret
+
+@dataclass
+class EventCondition(TrunkNode):
+    children: Tuple[Union[Sum, VarName], CompareSign, Union[Sum, VarName]] = field(repr=False)
+    
+    def get_left_var(self):
+        if type(self.children[0]) == VarName:
+            return self.children[0]
+        return None
+    
+    def get_left(self):
+        return self.children[0]
+    
+    def get_sign(self) -> CompareSign:
+        return self.children[1]
+    
+    def get_right_var(self):
+        if type(self.children[2]) == VarName:
+            return self.children[2]
+        return None
+    
+    def get_right(self):
+        return self.children[2]
+    
+    def to_string(self):
+        if type(self.get_left()) == VarName:
+            left_text = self.get_left().get_name_text()
+        else:
+            left_text = self.get_left().to_string()
+        sign = str(self.get_sign().get_sign())
+        if type(self.get_left()) == VarName:
+            right_text = self.get_right().get_name_text()
+        else:
+            right_text = self.get_right().to_string()
+        return left_text + ' ' + sign + ' ' + right_text
 
+@dataclass
+class EventConditionList(TrunkNode):
+    def get_all_conditions(self) -> List[EventCondition]:
+        return cast(List[EventCondition], self.children[1::4])
+
+    def get_all_parens(self) -> List[Parenthesis]:
+        return cast(List[Parenthesis], self.children[::2])
+    
+    def to_string(self) -> str:
+        ret = ''
+        for sym in self.children:
+            if isinstance(sym, EventCondition):
+                ret += sym.to_string()
+            elif sym is None:
+                continue
+            elif isinstance(sym, ParenthesisList):
+                ret += sym.to_string()
+            elif isinstance(sym, Logical):
+                ret += ' ' + sym.get_logic_symbol() + ' '
+            else:
+                ret += sym.to_string()
+        return ret
+    
+@dataclass
+class EventTrigger(TrunkNode):
+    children: Tuple[Keyword, Operator, Union[Boolean, Sum, Number]] = field(repr=False)
+    def get_keyword(self):
+        return self.children[0]
+    
+    def get_trigger(self):
+        return self.children[2]
+    
+    def get_trigger_str(self):

```

Comment:
consider using "trigger = self.get_trigger()" before the "if" and then reference trigger in each of the "isinstance".